runner/execer/os/os_execer.go

```diff
@@ -198,27 +220,26 @@ func (p *osProcess) Abort() (result execer.ProcessStatus) {
 }
 
 func memUsage(pid int) (execer.Memory, error) {
-	// Query for all sets of (pid, ppid, rss) and create a graph. Given a pid, walk the graph accruing memory.
+	// Query for all sets of (pid, sid, rss). Given a pid, find its associated sid.
+	// From there, sum the memory of all processes with the same session id.
 	// Note: look at the revision history for previous approaches which seemed inaccurate at times? ex:
 	//   str := "export P=%d; echo $(ps -orss= -p$(echo -n $(pgrep -g $P | tr '\n' ',')$P) | tr '\n' '+') 0 | bc"
 	str := `
 PID=%d
-PSLIST=$(ps -e -o pid= -o ppid= -o rss= | tr '\n' ';' | sed 's,;$,,')
+PSLIST=$(ps -e -o pid= -o sess= -o rss= | tr '\n' ';' | sed 's,;$,,')
```

Comment:
I don't know that we want to kill by sid on Linux, that will kill our entire worker. It will restart on a new host, but we still have to deal with the cost of a new startup.

Currently looking into the tests.